### ISSUE TITLE
No type checking for check-ed error type vs enclosing invokable's error type

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#19299]
**Description:**
$title.

**Steps to reproduce:**
Compilation should fail for
```ballerina
const R1 = "first reason";
const R2 = "second reason";

type E1 error<R1>;
type E2 error<R2>;

function foo() returns E1? {
    int i = check bar(); // should fail - check returns E2, but return type is E1?
}

function bar() returns int|E2 {
    return E2();
}

public function main() {
    E1? x = foo();
}
```

Currently fails at runtime:
```cmd
$ ballerina run xyz.bal 
Compiling source
        xyz.bal

Generating executables
Running executables

error: second reason 
        at xyz:bar(xyz.bal:12)
           xyz:foo(xyz.bal:8)
           xyz:main(xyz.bal:16)
```

**Affected Versions:**
```cmd
$ ballerina -v
Ballerina 1.0.1
Language specification 2019R3
```

**Comment by hasithaa** [comment]
Resolved with #19374

**Comment by MaryamZi** [comment]
Re-opening since the fix is being modified with https://github.com/ballerina-platform/ballerina-lang/pull/19529 to make this a warning instead of an error, for the 1.0.2 patch release.

Keeping this issue open to track making it an error.
